src/miniscript/limits.rs

```diff
@@ -50,3 +50,6 @@ pub const MAX_BLOCK_WEIGHT: usize = 4000000;
 /// Maximum pubkeys as arguments to CHECKMULTISIG
 // https://github.com/bitcoin/bitcoin/blob/6acda4b00b3fc1bfac02f5de590e1a5386cbc779/src/script/script.h#L30
 pub const MAX_PUBKEYS_PER_MULTISIG: usize = 20;
+
+/// Maximum TapLeafs allowed in a compiled TapTree
+pub const MAX_TAPLEAFS_PER_TAPTREE: usize = 10_000;
```

Comment:
Rename this constant to `MAX_COMPILATION_LEAVES`. As this constant is exposed publicly, others importing it might be confused `MAX_TAPLEAFS_PER_TAPTREE = 10000`  and assume this as maximum leaves in a taptreee. 

This is an artificial limitation of the compiler. 1) We should not expose it as `pub` and 2) Rename it as suggested above